public/app/plugins/datasource/prometheus/components/monaco-query-field/getOverrideServices.ts

```diff
@@ -0,0 +1,115 @@
+// also opens automatically.
+// but there is no API to achieve it.
+// the way to do it is to implement the `storageService`
+// interface, and provide our custom implementation,
+// which will defeault to `true` for the correct string-key.
+// unfortunately, while the typescript-interface exists,
+// it is not exported from monaco-editor,
+// so we cannot rely on typescript to make sure
+// we do it right. all we can do is to manually
+// lookup the interface, and make sure we code our code right.
+// our code is a "best effort" approach,
+// i am not 100% how the `scope` and `target` things work,
+// but so far it seems to work ok.
+// i would use an another approach, if there was one available.
+
+function makeStorageService() {
+  // we need to return an object that fullfills this interface:
+  // https://github.com/microsoft/vscode/blob/ff1e16eebb93af79fd6d7af1356c4003a120c563/src/vs/platform/storage/common/storage.ts#L37
+  // unfortunately it is not export from monaco-editor
+
+  const strings = new Map<string, string>();
+
+  // we want this to be true by default
+  strings.set('expandSuggestionDocs', true.toString());
+
+  return {
+    // we do not implement the on* handlers
+    onDidChangeValue: (data: unknown): void => undefined,
+    onDidChangeTarget: (data: unknown): void => undefined,
+    onWillSaveState: (data: unknown): void => undefined,
+
+    get: (key: string, scope: unknown, fallbackValue?: string): string | undefined => {
+      return strings.get(key) ?? fallbackValue;
+    },
+
+    getBoolean: (key: string, scope: unknown, fallbackValue?: boolean): boolean | undefined => {
+      const val = strings.get(key);
+      if (val !== undefined) {
+        // the interface-docs say the value will be converted
+        // to a boolean but do not specify how, so we improvise
+        return val === 'true';
+      } else {
+        return fallbackValue;
+      }
+    },
+
+    getNumber: (key: string, scope: unknown, fallbackValue?: number): number | undefined => {
+      const val = strings.get(key);
+      if (val !== undefined) {
+        return parseInt(val, 10);
+      } else {
+        return fallbackValue;
+      }
+    },
+
+    store: (
+      key: string,
+      value: string | boolean | number | undefined | null,
+      scope: unknown,
+      target: unknown
+    ): void => {
+      // the interface-docs say if the value is nullish, it should act as delete
+      if (value === null || value === undefined) {
+        strings.delete(key);
+      } else {
+        strings.set(key, value.toString());
+      }
+    },
+
+    remove: (key: string, scope: unknown): void => {
+      strings.delete(key);
+    },
+
+    keys: (scope: unknown, target: unknown): string[] => {
+      return Array.from(strings.keys());
+    },
+
+    logStorage: (): void => {
+      console.log('storage log');
+    },
+
+    migrate: (): Promise<void> => {
+      // we do not implement this
```

Comment:
What's the consequence of not implementing this (and methods below)? As in, what could possibly go wrong in the future? ;)

Reply:
i have absolutely no idea :-) 

the only source of info we have is the comments in the interface-definition-file ( https://github.com/microsoft/vscode/blob/ff1e16eebb93af79fd6d7af1356c4003a120c563/src/vs/platform/storage/common/storage.ts#L37 ), and the fact that it does not crash when we use the monaco-query-editor.

for `migrate`:

i looked through the visual-studio-code, and all i find are instances like:
```typescript
	async migrate(toWorkspace: IWorkspaceInitializationPayload): Promise<void> {
		throw new Error('Migrating storage is currently unsupported in Web');
	}
```
and
```typescript
	async migrate(toWorkspace: IWorkspaceInitializationPayload): Promise<void> {
		// not supported
	}
```

so i THINK maybe it's ok to not implement it.

for `isNew`:

the interface-docs say it should return whether it was created in this "session" or not. so i changed the code to always return true, because we do not persist it. i'm not sure of course if this is the right way, but it's my best guess. i checked the usage in the vscode source code, and there is a storage-thing that simply stores it in a value and returns it, and it is set only from the electron-version of the code, based on some logic.

for `flush`:
the interface-docs say it is so that the storage can flush it's state, for example when a shutdown will happen. i don't want to persist anything, so i just went with not doing anything there.

in short, this is not the nicest code we have here, unfortunately there is no other way to achieve what we need (auto-open extra-help window) :-(

Reply:
Thanks for more info!